# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/l10n/FluentBundle.cpp
# Commit: 92b87b0f776c
# Full Hash: 92b87b0f776c3049499038d380190351e2371324
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-04 03:56:33
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part2: Switch Localization class to use localization-ffi. r=emilio,nika
#   
#   Depends on D104788
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104789
# ==============================================================================

diff -r 878b052ebeaa -r 92b87b0f776c intl/l10n/FluentBundle.cpp
--- a/intl/l10n/FluentBundle.cpp	Tue Aug 03 16:25:10 2021 +0000
+++ b/intl/l10n/FluentBundle.cpp	Tue Aug 03 16:25:10 2021 +0000
@@ -188,8 +188,10 @@
   return true;
 }
 
-static void ConvertArgs(const L10nArgs& aArgs,
-                        nsTArray<ffi::L10nArg>& aRetVal) {
+/* static */
+void FluentBundle::ConvertArgs(const L10nArgs& aArgs,
+                               nsTArray<ffi::L10nArg>& aRetVal) {
+  aRetVal.SetCapacity(aArgs.Entries().Length());
   for (const auto& entry : aArgs.Entries()) {
     if (!entry.mValue.IsNull()) {
       const auto& value = entry.mValue.Value();